binder/src/androidTest/java/io/grpc/binder/internal/BinderClientTransportTest.java

```diff
@@ -127,39 +124,51 @@ public void setUp() throws Exception {
             .build();
 
     serverAddress = HostServices.allocateService(appContext);
-    HostServices.configureService(serverAddress,
+    HostServices.configureService(
+        serverAddress,
         HostServices.serviceParamsBuilder()
-          .setServerFactory((service, receiver) ->
-              BinderServerBuilder.forAddress(serverAddress, receiver)
-                .addService(serviceDef)
-                .build())
-          .build());
-
-    transport =
-        new BinderTransport.BinderClientTransport(
-            appContext,
-            serverAddress,
-            BindServiceFlags.DEFAULTS,
-            ContextCompat.getMainExecutor(appContext),
-            executorServicePool,
-            executorServicePool,
-            SecurityPolicies.internalOnly(),
-            InboundParcelablePolicy.DEFAULT,
-            Attributes.EMPTY);
-
-    Runnable r = transport.start(transportListener);
-    r.run();
-    transportListener.awaitReady();
+            .setServerFactory(
+                (service, receiver) ->
+                    BinderServerBuilder.forAddress(serverAddress, receiver)
+                        .addService(serviceDef)
+                        .build())
+            .build());
+  }
+
+  private class BinderClientTransportBuilder {
```

Comment:
FYI: in some cases we have found it useful to use the transport's ChannelBuilder and call `buildClientTransportFactory()` from the test instead of the normal `build()`. It avoids repeating some defaults but also can prevent needing yet-another-builder. For that, you expose `buildClientTransportFactory()` as package-private, so it isn't available for others. I don't know if that pattern is helpful for these tests, but wanted to mention it.

Reply:
OK I was able to locate this pattern in the unit tests for the other transports. Thanks for mentioning it.

Agreed this builder duplicates some logic from the real BinderChannelBuilder. OTOH, it lets me include only the transport properties that are relevant to each test case (go/tott-529). Calling newClientTransport() would force every test case to specify such as the destination address, ClientTransportOptions and ChannelLogger even when they aren't relevant. On balance I think yet another builder is the least bad but LMK if you feel strongly.

Reply:
Don't change it for me. I was just mentioning it was an option.

Something isn't adding up... the options and logger would be annoying. It was easier than that... Oh! I guess what I'm thinking of is applying the "pass Builder to constructor" pattern, and we don't (yet) use it for any of the transports. [ManagedChannelImpl uses it](https://github.com/grpc/grpc-java/commit/0a8d761c199eb5a02e683adaa922e0e9a6773012). The pattern normally works better because Builder is a nested class and so private fields are accessible (although some people may choose to make them package-private to avoid synthetic methods). For gRPC it isn't perfect because all the fields need to become package-private, but as the number of fields increases the value increases.

Reply:
👍